### ISSUE TITLE
DefaultCompactionPlanner not seeing updated deprecated properties

### ISSUE THREAD

**Issue by dlmarion** [apache/accumulo#3383]
**Describe the bug**
Changing `tserver.compaction.major.thread.files.open.max` in 2.1 requires a tserver bounce to take effect. User remote debugged the tserver and found that [configs](https://github.com/apache/accumulo/blob/2.1/core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionServicesConfig.java#L114) only contained the new properties, not the deprecated ones.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: 2.1.0

**Expected behavior**
As the new and old properties are marked as mutable, would expect that changes to the old deprecated properties would have an effect.

**Additional context**
Workaround: use new properties. If using old properties, user may not be able to determine if changes to properties values are taking effect.

**Comment by EdColeman** [comment]
Background:  During the upgrade process, properties are processed, renamed and the stored in [DeprecatedPropertyUtil](https://github.com/apache/accumulo/blob/e46cd0e72c366ce9873989930dc25d15e82d90f1/core/src/main/java/org/apache/accumulo/core/conf/DeprecatedPropertyUtil.java#L36)  One interesting thing is the javadoc for [getReplacementName](https://github.com/apache/accumulo/blob/e46cd0e72c366ce9873989930dc25d15e82d90f1/core/src/main/java/org/apache/accumulo/core/conf/DeprecatedPropertyUtil.java#L83C25-L83C25) says:
```
This is expected to be used only with system properties stored in the SiteConfiguration and ZooConfiguration, and not for per-table or per-namespace configuration in ZooKeeper.
```
With the property conversion into a single node, all ZooKeeper properties are processed and renamed, including per-namespace and per-table.

So that there is one canonical source / value  it seems desirable to keep the properties "normalized" with a single name (ideally the "new" name) otherwise there could be issues with property hierarchy inheritance.  Using the example `tserver.compaction.major.thread.files.open.max` - the new name is `tserver.compaction.major.service.default.planner.opts.maxOpen` if this was set as a system property it would be converted to the new name. If it was overridden on a table and *NOT* converted, what name would be the correct value to apply to the table?  Performing the conversion removes this ambiguity.

To keep a canonical name (the new name) there are (at least) two possible approaches. 
  1) Perform the check / conversion each time the property is set and display a warning that the conversion occurred.
  2) Fail the set with an appropriate warning.

I'll take a run at option 1 unless someone has a better path forward.

**Comment by EdColeman** [comment]
The previous comment was based on the thought that the rename was handling more that the manager conversion - more investigation is needed. 

From a quick look, both version of the compaction file open max are present as a default, and both can be set as a system property, so not sure why the update would not be visible.

**Comment by ctubbsii** [comment]
I think this is probably sufficiently resolved by #3680 , but can be reopened if there's more to be done here.
